File: sdk/typescript/src/namespaces/stats.ts
import type {
  GetStatsParams,
  GetWrappedParams,
} from "../generated/types.js";
import type { RequestOptions } from "../types.js";
import type { Call } from "./_helpers.js";

export class StatsNamespace {
  constructor(private readonly call: Call) {}

  getStats(params: GetStatsParams, opts?: RequestOptions) {
    return this.call("app.rocksky.stats.getStats", "GET", {
      params,
      ...opts,
    });
  }

  getWrapped(
    params: GetWrappedParams,
    opts?: RequestOptions,
  ) {
    return this.call("app.rocksky.stats.getWrapped", "GET", {
      params,
      ...opts,
    });
  }
}

File: sdk/go/rocksky/stats.go
package rocksky

import "context"

// StatsService groups endpoints under app.rocksky.stats.*.
type StatsService struct{ c *Client }

// GetStatsParams identifies the user whose totals are returned.
type GetStatsParams struct {
	Actor string // handle or DID, required
}

// GetStats returns aggregate counters (total scrobbles, distinct artists, etc.).
// XRPC: app.rocksky.stats.getStats.
func (s *StatsService) GetStats(ctx context.Context, p GetStatsParams) (*StatsView, error) {
	q := newQuery()
	q.setString("did", p.Actor)
	out := &StatsView{}
	if err := s.c.query(ctx, "app.rocksky.stats.getStats", q.Values(), out); err != nil {
		return nil, err
	}
	return out, nil
}

// GetWrappedParams selects the year to summarize.
type GetWrappedParams struct {
	Actor string // handle or DID, required
	Year  int    // 0 means the most recent full year
}

// GetWrapped returns the year-end recap for a user.
// XRPC: app.rocksky.stats.getWrapped.
func (s *StatsService) GetWrapped(ctx context.Context, p GetWrappedParams) (*WrappedView, error) {
	q := newQuery()
	q.setString("did", p.Actor)
	q.setInt("year", p.Year)
	out := &WrappedView{}
	if err := s.c.query(ctx, "app.rocksky.stats.getWrapped", q.Values(), out); err != nil {
		return nil, err
	}
	return out, nil
}
